File: Problems/Button with Longest Push Time.py
class Solution:
    def buttonWithLongestTime(self, events: List[List[int]]) -> int:
        
        max_diff = events[0][1]
        output = events[0][0]

        for i in range(1, len(events)):

            time_diff = events[i][1] - events[i - 1][1]
            
            if time_diff > max_diff:
                max_diff = time_diff
                output = events[i][0]

            elif max_diff == time_diff:
                output = min(output, events[i][0])


        return output

File: Problems/Button with Longest Push Time.cpp
class Solution {
public:
    int buttonWithLongestTime(vector<vector<int>>& events) {
        int maxi = events[0][0], maxdur = events[0][1];
        for(int i = 1, n = events.size(); i < n; ++i){
            int idx = events[i][0], dur = events[i][1] - events[i - 1][1];
            if(dur > maxdur){
                maxdur = dur;
                maxi = idx;
            }
            else if(dur == maxdur) maxi = min(maxi , idx);
        }
        return maxi;
    }
};
